docs/change-detection.md

```diff
@@ -3,22 +3,73 @@
 When changing your infrastructure (made up of a set of stacks) it's common to
 make several changes to several stacks. But now that you have multiple terraform
 states (per stack), how to apply the changes only to the affected resources?
+Keep in mind that we don't want to just blindly execute _plan/apply_ to the
+unchanged stacks to reduce the blast radius.
 
-Terramate comes with support for change detection by imposing the following workflow:
+We solve that by leveraging the power of the VCS (Version Control System)
+already in place. At the moment, Terramate only supports `git` but other VCSs
+can be added in the future.
 
-1. The default branch (commonly `main`) is considered to be the stable branch that represents the deployed state of our IaC.
-2. Changes that should be planned and applied should be added through a feature or bugfix branch.
+So the idea is as simple as computing `git diff` changes between the revision of
+the last `terraform applied` change (ie. the released revision) and the current
+change. Let's call the released revision of `baseref`, which means
+`base reference` which commonly is default branch (`origin/main` or
+`origin/default`) and this is the reference we're going to compare when
+computing the changed stacks.
```

Comment:
```suggestion
`origin/default`).
```

This seemed a little redundant :thinking:

Reply:
yes, I think I'm gonna change the text to the below:

```
The approach is as simple as computing the changed stacks from the changed files 
discovered by the `git diff` between the revision of the last `terraform applied` 
change (ie. the released revision) and the current change. Let's call the released 
revision `baseref`, which means `base reference` which commonly is the default 
branch (`origin/main` or `origin/default`)
```
because that was the only part talking about computing changed stacks from the diff, all the rest talked about diffing files only.

wdyt?

Reply:
Actually, the `baseref` naming can go in a separate phrase now:

```
The approach is as simple as computing the changed stacks from the changed files 
discovered by the `git diff` between the revision of the last `terraform applied` 
change (ie. the released revision) and the current change.

Let's call the released revision `baseref`, which means `base reference` which
commonly is the default branch (`origin/main` or `origin/default`)
```

Reply:
Liked the last one :100: